plugin/bookmark.vim

```diff
@@ -254,46 +254,138 @@ function! CallDeprecatedCommand(fun, args)
 endfunction
 
 function! BookmarkModify(diff)
+  let current_line = line('.')
+  let new_line_nr = current_line + a:diff
+  call BookmarkMoveToLine(new_line_nr)
+endfunction
+
+
+
+function! BookmarkMoveToLine(target)
   call s:refresh_line_numbers()
+
   let file = expand("%:p")
   if file ==# ""
     return
   endif
 
   let current_line = line('.')
+  if a:target == current_line
+    return
+  endif
+
   if bm#has_bookmark_at_line(file, current_line)
-    let new_line_nr = current_line + a:diff
-    if bm#has_bookmark_at_line(file, new_line_nr)
+    "ensure we're trying to move to a valid line nr
+    let max_line_nr = line('$')
+    if a:target <= 0 || a:target > max_line_nr
+      echo "Can't move bookmark beyond file bounds"
+      return
+    endif
+    if bm#has_bookmark_at_line(file, a:target)
       echo "Hit another bookmark"
       return
     endif
-    
-
 
     let bookmark = bm#get_bookmark_by_line(file, current_line)
-    call bm_sign#update_at(file, bookmark['sign_idx'], new_line_nr, bookmark['annotation'] !=# "")
+    call bm_sign#update_at(file, bookmark['sign_idx'], a:target, bookmark['annotation'] !=# "")
 
     let bufnr = bufnr(file)
-    let line_content = getbufline(bufnr, new_line_nr)
+    let line_content = getbufline(bufnr, a:target)
     let content = len(line_content) > 0 ? line_content[0] : ' '
-    call bm#update_bookmark_for_sign(file, bookmark['sign_idx'], new_line_nr, content)
-
-    call cursor(new_line_nr, 1)
-    execute ":redraw!"
+    call bm#update_bookmark_for_sign(file, bookmark['sign_idx'], a:target, content)
+    call cursor(a:target, 1)
+    " redraw!
```

Comment:
Is a redraw really needed? Had to disable as it would cause vspec to fail https://github.com/kana/vim-vspec/issues/41